plugins/notifications-backend/src/service/router.ts

```diff
@@ -86,6 +87,36 @@ export async function createRouter(
       .catch(next);
   });
 
+  router.put('/notifications/read', (request, response, next) => {
+    const { messageID, user, read } = request.query;
+    if (
+      typeof messageID !== 'string' ||
+      typeof user !== 'string' ||
+      typeof read !== 'string'
+    ) {
+      throw new Error(
+        'the following query parameters must be provided: messageID - string, user - string, read - 0/1 (boolean)',
+      );
+    }
+
+    let readBool: boolean;
+
+    switch (read) {
+      case '1':
+        readBool = true;
+        break;
+      case '0':
+        readBool = false;
+        break;
+      default:
+        throw new Error('value of parameter "read" must be either "0" or "1"');
+    }
+
+    setRead(dbClient, messageID, user, readBool)
+      .then(result => response.json(result))
```

Comment:
We are not sending any data here, the handler's return type is `void`.
We should use
```
.then(() => response.end())
```

Reply:
done